### ISSUE TITLE
prevent adding buffer to Nerdtree window

### ISSUE THREAD

**Issue by izelnakri** [scrooloose/nerdtree#930]
Following command seems to be very useful for preventing people to open a buffer in nerdtree window:

```vim
function! move_window_when_in_nerd_tree()
  if (expand('%') =~# 'nerd_tree' && winnr('$') > 1)
    exe "normal! \<c-w>\<c-w>"
  endif
endfunction

" this prevents opening the new buffer in nerdtree window:
" happens when people want to move to a file with fzf, search & move with ripgrep, :e and more:
au BufAdd * call move_window_when_in_nerd_tree()
```

**Comment by PhilRunninger** [comment]
@izelnakri , this is not working as advertised. There are a few things wrong with it.
1. The function name must be capitalized or start with `s:`.
2. The regex comparison needs to be case insensitive - `=~?`
3. According to the `:h BufAdd` documentation, `%` may be a different buffer than you expect it to be.

I made the first two changes locally and executed these commands:
```
$ vim vimrc
:NERDTree
```
This caused there to be two windows containing the NERDTree buffer. Things got even stranger after that.

Your intent is a good one; the window containing NERDTree shouldn't switch buffers. This, of course, is a user preference, and I won't be adding it as a feature in NERDTree.

Instead, it should be part of one's vimrc. Here is the way I implemented it:
```
" If previous buffer was NERDTree, go back to it.
autocmd BufEnter * if bufname('#') =~# "^NERD_tree_" | b# | endif
```

**Comment by izelnakri** [comment]
Thank you very much for your thoughtful response! 

I'll try your suggestion, I did experience the occasional weird behavior you've just mentioned. Mine was certainly a hacky implementation I found somewhere else, I'll try your suggestion and let you know. 

I think we should add this to README as a suggestion even if it wont be in the plugin. I suppose this is a common annoyance many people experience.
